src/zarr/v3/group.py

```diff
@@ -271,8 +290,32 @@ def __repr__(self):
     async def nchildren(self) -> int:
         raise NotImplementedError
 
-    async def children(self) -> AsyncIterator[AsyncArray, AsyncGroup]:
-        raise NotImplementedError
+    async def children(self) -> AsyncGenerator[AsyncArray, AsyncGroup]:
+        """
+        Returns an AsyncGenerator over the arrays and groups contained in this group.
+        This method requires that `store_path.store` supports directory listing.
+        """
+        if not self.store_path.store.supports_listing:
+            msg = (
+                f"The store associated with this group ({type(self.store_path.store)}) "
+                "does not support listing, "
+                "specifically via the `list_dir` method. "
+                "This function requires a store that supports listing."
+            )
+
+            raise ValueError(msg)
+        subkeys = await self.store_path.store.list_dir(self.store_path.path)
+        # would be nice to make these special keys accessible programmatically,
+        # and scoped to specific zarr versions
+        subkeys_filtered = filter(lambda v: v not in ("zarr.json", ".zgroup", ".zattrs"), subkeys)
+        # is there a better way to schedule this?
+        for subkey in subkeys_filtered:
+            try:
+                yield await self.getitem(subkey)
+            except KeyError:
+                # keyerror is raised when `subkey``names an object in the store
+                # in which case `subkey` cannot be the name of a sub-array or sub-group.
+                pass
```

Comment:
I might be missing something here but if `subkey` was just found in the store, why would we be getting a `KeyError`? This `pass` seems like it could silently ignore something that we care about.

Reply:
If `"foo"` is the name of a random non-zarr-related file in the store, `group.getitem("foo")` raises `KeyError`, because `"foo"` is not the name of a sub-array or sub-group. So in this case, a `KeyError` just means there's a random file in there, that we want to ignore. I'm not sure that there are sources of `KeyError` (or any other exception) that we would want to handle here?